src/main/java/com/limechain/client/FullNode.java

```diff
@@ -242,4 +264,54 @@ TrieStructure<Pair<Optional<byte[]>, Optional<byte[]>>> initializeDatabaseStorag
 
         return trieStructure;
     }
+
+    /**
+     * Inserts trie nodes into the key-value repository.
+     *
+     * @param db The key-value repository where trie nodes are to be stored.
+     * @param insertTrieNodes The list of trie nodes to be inserted.
+     * @param entriesVersion The version number of the trie entries.
+     */
+    private void insertStorage(KVRepository<String, Object> db, List<InsertTrieNode> insertTrieNodes, byte entriesVersion) {
+        try {
+            for (InsertTrieNode trieNode : insertTrieNodes) {
+                //Insert tn(trie node) to db
+                db.save("tn:" + new String(trieNode.getMerkleValue()), trieNode.getPartialKeyNibbles());
+
+                // Handle storage value
```

Comment:
Some of the comments in this block can be removed and the code extracted in private methods.